src/lightning/pytorch/core/optimizer.py

```diff
@@ -382,7 +382,7 @@ def step(self, closure: Optional[Callable] = None) -> None:
         if closure is not None:
             closure()
 
-    def zero_grad(self, set_to_none: Optional[bool] = False) -> None:
+    def zero_grad(self, set_to_none: Optional[bool] = True) -> None:
```

Comment:
Even though it's fine to update this, it won't have any performance implication. This class is just a mock we use when `configure_optimizers` is not defined (generally because the Trainer is evaluating)

Lightning just uses the optimizer returned from `configure_optimizers` when training

Reply:
Thanks for the quick response! Does this mean lightning already defaults to set_to_none=True? If so, that's great and this PR isn't quite necessary.

Reply:
I'm also fine with merging this.

Reply:
Sounds good, I'll clean up the PR description to say so. Thanks for your input everyone!